DataStructures/BitList.cs

```diff
@@ -0,0 +1,1753 @@
+            if (IsEmpty)
+            {
+                return new decimal[0];
+            }
+            else
+            {
+                try
+                {
+                    var longs = ToInt64Array().ToList();
+                    if (longs.Count % 2 != 0)
+                    {
+                        longs.Add(0);
+                    }
+
+                    var decimals = new List<decimal>();
+                    for (var i = 0; i < longs.Count; i += 2)
+                    {
+                        decimals.Add(new decimal(new BitList(new[] { longs[i], longs[i + 1] }).ToInt32Array()));
+                    }
+
+                    return decimals.ToArray();
+                }
+                catch
+                {
+                    throw new ConversionErrorException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="char"/> array.
+        /// </summary>
+        /// <returns>A <see cref="char"/> array.</returns>
+        public char[] ToCharArray()
+        {
+            //Convert the bits field (BitArray) to Int16 array and then to a Char array.
+            if (IsEmpty)
+            {
+                return new char[0];
+            }
+            else
+            {
+                var shorts = ToInt16Array();
+                var chars = new List<char>();
+                foreach (var bits in shorts)
+                {
+                    chars.Add((char)bits);
+                }
+
+                return chars.ToArray();
+            }
+        }
+
+        /// <summary>
+        /// Convert this instance to a <see cref="string"/> array.
+        /// </summary>
+        /// <returns>A <see cref="string"/> array.</returns>
+        public override string ToString()
+        {
+            //Convert the bits field (BitArray) to a Char array and then to a String.
+            if (IsEmpty)
+            {
+                return string.Empty;
+            }
+            else
+            {
+                return new string(ToCharArray());
+            }
+        }
+
+        /// <summary>
+        /// Return true if the specified BitList is equal to this istance.
+        /// </summary>
+        /// <param name="obj">A list of bit.</param>
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as BitList);
+        }
+
+        /// <summary>
+        /// Gets hash code.
+        /// </summary>
+        /// <returns>Hash code.</returns>
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+
+        IEnumerator<bool> IEnumerable<bool>.GetEnumerator()
+        {
+            return ToBooleanArray().ToList().GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return ToBooleanArray().ToList().GetEnumerator();
+        }
+
+        private class Unmanaged<T> where T : unmanaged { }
```

Comment:
Codacy found an issue: ['T' is not used in the class.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)